### DISCUSSION TITLE
Combining different CVA functions

### DISCUSSION THREAD

**Discussion by matibenav** [joe-bell/cva#188] [Ideas]
Hi, dear reader.
I was thinking about composable functions, as I have a use case at my workplace: I have a `Pill` component that sometimes acts like a Button and sometimes like a span. This can be achieved by just interpolating strings, but I was thinking that maybe something else can be done that can provide typing when needed.

I did a _short and basic_ combining function, as I don't have access to the typings from CVA (And I don't think I should copy/paste them in my repo, because _maintenance_)

```ts
export function combineStylesFunctions<P>(...funcs: Array<(props?: P) => string>) {
  return (props: P): string => funcs.map((func) => func(props)).join(' ');
}
```

The thing is that this will not provide the desirable typing when applying VariantProps. 
```ts
type Props = VariantProps<typeof YOUR_STYLE_FUNCTION>;
```
Which should be the typing of the intersection of the combined functions. 

Not sure how it could be implemented, but I feel it would be a nice feature to have in order to ensure composition.

**Comment by joe-bell** [comment]
This is great, thanks for sharing! I'll pop it in my `cva@1.0` consideration list

**Comment by joe-bell** [comment]
https://github.com/joe-bell/cva/pull/193
